nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java

```diff
@@ -0,0 +1,300 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.fixserialization;
+
+import com.google.common.base.Preconditions;
+import com.sun.source.util.Trees;
+import com.uber.nullaway.fixserialization.qual.AnnotationFactory;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import javax.lang.model.element.Element;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+/** Config class for Fix Serialization package. */
+public class FixSerializationConfig {
+
+  /**
+   * If activated, for all reporting errors, NullAway will serialize information and suggests type
+   * changes to resolve them, in case these errors could be fixed by adding a @Nullable annotation.
+   * These data will be written at output directory.
+   */
+  public final boolean suggestEnabled;
+  /**
+   * If activated, serialized information of a fix suggest will also include the enclosing method
+   * and class of the element involved in error.
+   */
+  public final boolean suggestEnclosing;
+  /** If activated, NullAway will write reporting errors in output directory. */
+  public final boolean logErrorEnabled;
+  /** If activated, errors information will also include the enclosing method and class. */
+  public final boolean logErrorEnclosing;
+  /** The directory where all files generated/read by Fix Serialization package resides. */
+  public final String outputDirectory;
+
+  public final AnnotationFactory annotationFactory;
+  /**
+   * Contains a set of classes fully qualified name. If not empty, NullAway will only report errors
+   * in these classes.
+   */
+  public final Set<String> workList;
+
+  public final Writer writer;
+
+  public FixSerializationConfig() {
+    suggestEnabled = false;
+    suggestEnclosing = false;
+    logErrorEnabled = false;
+    logErrorEnclosing = false;
+    annotationFactory = new AnnotationFactory();
+    // No class is excluded.
+    workList = Collections.singleton("*");
+    outputDirectory = null;
+    writer = null;
+  }
+
+  /**
+   * * Sets all flags based on their values in {@code outputdierectory/explorer.config} json file.
+   *
+   * @param autofixEnabled Flag value in main NullAway config. If false, all flags here will be set
+   *     to false.
+   * @param outputDirectory Directory where all files generated/read by Fix Serialization package
+   *     resides.
+   */
+  public FixSerializationConfig(boolean autofixEnabled, String outputDirectory) {
+    // if autofixEnabled is false, all flags will be false regardless of their given value in json
+    // config file.
+    Preconditions.checkNotNull(outputDirectory);
+    JSONObject jsonObject = null;
```

Comment:
It looks like JSON is only used for the configuration file.  Hard to believe I'm saying this, but could we use XML instead?  Then we could use parsers built into the Java standard library ([javax.xml.parsers](https://docs.oracle.com/javase/8/docs/api/index.html?javax/xml/parsers/package-summary.html)) and wouldn't need to add a third-party dep just for parsing.

Reply:
4eb83ddfdcb282861cb9c624a258c5f6e1e0fd2f

I converted ```json``` files to ```xml```. Below is the format we expect to read for ```Serialization Config```:

```xml
<serialization>
    <error activation="true" enclosing="true"/>
    <suggest activation="true" enclosing="true"/>
    <annotation>
        <nullable>javax.annotation.Nullable</nullable>
        <nonnull>javax.annotation.Nonnull</nonnull>
    </annotation>
</serialization>
```

Reply:
Since we are taking in a configuration file, why not just have the output directory be specified in the config file?

Reply:
When output directory is specified for ```NullAway``` it will look for the config file at ```output_dir/serializer.xml```. If we want to specify the output dir in config file, then we should give ```NullAway``` the path to config file. Please let me know if you would rather the second option. Thank you.

Reply:
Hi @msridhar, I believe this is ready for another review. With best regards.

Reply:
> When output directory is specified for `NullAway` it will look for the config file at `output_dir/serializer.xml`. If we want to specify the output dir in config file, then we should give `NullAway` the path to config file. Please let me know if you would rather the second option. Thank you.

Yes, we should do the second option.  Expecting the configuration file inside the output directory is a bit strange.

Reply:
I changed it to accept ```config path``` and the ```outputdir``` is given in the config path.
7771f125d89722301816e23ff2cda206c23abbfe